acceptance/indy_acceptance/test_scripts/test_scenario_base.py

```diff
@@ -0,0 +1,107 @@
+"""
+Created on Nov 22, 2017
+
+@author: khoi.ngo
+
+Containing the test base class.
+"""
+import asyncio
+import inspect
+import os
+import time
+
+from libraries import common
+from libraries import constant
+from libraries import utils
+from libraries.constant import Color
+from libraries.logger import Logger
+from libraries.result import TestResult, Status
+from libraries.step import Steps
+from libraries.utils import generate_random_string, run_async_method,\
+                            make_final_result
+
+
+class TestScenarioBase():
+    """
+    Test base....
+    All test scenario should inherit from this class.
+    This class controls the work flow and hold some general test data for
+    test scenarios that inherit it.
+    """
+    pool_name = generate_random_string("test_pool")
+    wallet_name = generate_random_string("test_wallet")
+    pool_handle = 0
+    wallet_handle = 0
+    pool_genesis_txn_file = constant.pool_genesis_txn_file
+    logger = None
+    steps = None
+    test_result = None
+    test_name = ""
+
+    def init_data_test(self):
+        """
+        Init test data.
+        If the test case need some extra test date
+        then just override this method.
+        """
+        self.test_name = os.path.splitext(
+            os.path.basename(inspect.getfile(self.__class__)))[0]
+        self.test_result = TestResult(self.test_name)
+        self.steps = Steps()
+        self.logger = Logger(self.test_name)
+
+    async def execute_precondition_steps(self):
+        """
+         Execute pre-condition of test scenario.
+         If the test case need some extra step in pre-condition
+         then just override this method.
+        """
+        common.clean_up_pool_and_wallet_folder(
+            self.pool_name, self.wallet_name)
+
+    async def execute_postcondition_steps(self):
+        """
+        Execute post-condition of test scenario.
+        If the test case need some extra step in post-condition
+        then just override this method.
+        """
+        await common.clean_up_pool_and_wallet(self.pool_name, self.pool_handle,
+                                              self.wallet_name,
+                                              self.wallet_handle)
+
+    async def execute_test_steps(self):
+        """
+        The method where contain all main script of a test scenario.
+        All test scenario inherit TestScenarioBase have to override this method
+        """
+        pass
+
+    def execute_scenario(self):
+        """
+        Execute the test scenario and control the work flow of
+        this test scenario.
+        """
+        begin_time = time.time()
+        self.init_data_test()
+        utils.print_with_color("\nTest case: {} ----> started\n"
+                               .format(self.test_name), Color.BOLD)
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
```

Comment:
We need not to explicitly create an event loop. We should just call `asyncio.get_event_loop` to get the event loop for the current context using the default event loop policy.

Reply:
As running one test at a time, we will remove all new, set and close event loop.